### DISCUSSION TITLE
What happens if a third-party contributor changes .github/actions workflows?

### DISCUSSION THREAD

**Discussion by janniks** [community/community#33154] [Actions]
Assuming a fork which opens a PR back to the original repo.
In this case GitHub offers a `Approve and run` button to allow running the .github/actions from an outside contribution (e.g. first-time contributor).

❓ If a third-party contributor was "approved", do they need to be approved again for the next pull-request OR could they now theoretically change the .github/actions and thus steal secrets?

❓Are there special securities in place for changes in .github/actions? e.g. does GitHub actions use the workflow definitions from the fork or the base repo (in case of an outside branch coming in)?

> ⚠️ This is a large security concern for us and the current docs aren't very explicit. Other answers also don't go into detail of the "approve" button flow.

Thanks 🙏

---

**Sources:**
- [Approving workflow runs from public forks](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks)
- [Can pull requests change actions?](https://github.com/orgs/community/discussions/27084?#discussioncomment-3254544)

**Comment by jfullstackdev** [comment]
I'm not very sure of the behavior but in my experience, I was added to an Org (not a member) with write permission only, I can merge the commits without their approval,

so I think that should be the behavior of that, isn't it ?

**Reply by janniks** [reply]
In this case I mean a third-party contributor (no permisssions in org), but the `approve and run` button was clicked.

**Comment by airtower-luna** [comment]
You can configure what needs to be approved in the repository settings (under Actions > General):

![grafik](https://user-images.githubusercontent.com/5814919/190392933-975ad8f3-22e5-4512-b76c-f80ac581786d.png)

Note that `pull_request` events triggered from forks [don't receive secrets anyway](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflows-in-forked-repositories), approval or not. If you use other event types to run jobs with secrets on pull requests, see [Keeping your GitHub Actions and workflows secure Part 1: Preventing pwn requests](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/) for a deep dive into the security considerations and mitigations you should make.
